### ISSUE TITLE
Greenclip does not copy selection

### ISSUE THREAD

**Issue by yeungocanh** [erebe/greenclip#26]
Hello.
I used this command 
`rofi -modi "clipboard:greenclip print" -show clipboard -run-command '{cmd}'`
to show history and pressed `Enter` to select one.
But i have nothing in my clipboard and i can not paste to anywhere.
I use Arch linux and use pacaur to install greenclip.
Thanks

**Comment by erebe** [comment]
I am for 2 weeks in Nepal right now, I will get back to you when I come back to Paris.
In the mean time, try to do `greenclip print 'foobar'` and see if you have foobar in your clipboard.
Be sure you have greenclip's binary in your PATH

**Comment by erebe** [comment]
Hi back,

Have you been able to try out the command i gave you ?

**Comment by mac1202** [comment]
I had the same issue as OP, setting usePrimarySelectionAsInput to True improved the situation but it still broken for now. In some app it work properly (firefox) in other I can copy but cannot paste (konsole).

**Comment by erebe** [comment]
Are you using KDE ? 

if so, be aware that there is already a clipboard manager in KDE.
Follow the last comment of this post in order to disable it
https://www.reddit.com/r/unixporn/comments/5mszii/greenclip_simple_clipboard_manager_integrated/

**Comment by mac1202** [comment]
I have tested it on 2 different PC, one under kde the other under lxqt. I have disabled the previous clipboard manager ( klipper and clippit) on both of them, but have the same issue on both PC.

**Comment by erebe** [comment]
Can you try out the 3.0-beta version and tell me who own the selection 
https://github.com/erebe/greenclip/releases/download/3.0-beta/greenclip

You need to `rm -rf ~/.cache/greenclip*` and be sure that an other greenclip daemon is not running.
So just:

1. pkill -9 greenclip
2. rm -rf ~/.cache/greenclip*
3. run in a terminal greenclip daemon
4. Try to copy stuff and see in the output in the terminal who own the selection

**Comment by mac1202** [comment]
Every time I copy something greenclip output the following line

`Just ""`

**Comment by erebe** [comment]
Ok, So you don't any luck ;'(
The program that is steeling the selection does not expose its name.

Just to be sure, can you

**Comment by mac1202** [comment]
```
 mac@Laptop  ~  greenclip print "toto"
 mac@Laptop  ~  ps aux | grep greenclip
mac       3970  0.0  0.0 1073748388 3716 pts/2 S    15:40   0:00 greenclip print toto
mac       3989  0.0  0.0  10888  2268 pts/2    S+   15:40   0:00 grep --color=auto greenclip
```

**Comment by erebe** [comment]
With `usePrimarySelectionAsInput = False` in your config file ?

**Comment by erebe** [comment]
If you already have `usePrimarySelectionAsInput = False` in your config file.
Can you give me your 

1. rofi config 
2. greenclip config
3. OS and window manager

In order for me to reproduce your environnement and try to troubleshoot the issue

**Comment by mac1202** [comment]
I test the beta version you give me on both PC and itseem to work properly now. The only thing that miss to be perfect is on option to sync primary selection and  clipboard, most clipboard manager  have this on linux and it make usage a lot more convenient.

**Comment by erebe** [comment]
Ok good if things are working for you now.

Regarding the sync between primary and clipboard selections, it will not happen.
The reason is due to how greenclip is designed to interact with the clipboard.
In the Xorg protocol, a single application is responsible to own the clipboard and is responsible to handle the request to it from other applications.
Thus it put the responsability on each application to implement correctly all the possible way to request the content of the clipboard (support for strings, mimes types, convertion, chunked selection, ...)
There is no RFC to decribe all the possibilities and how thing works is mostly an had hoc knowledge.
Put it simply it is extremly complicated/lot of work to get it right or to handle all the cases correctly.

Now, the way most clipboard managers are working is by stealing the selection ownership from other applications. Meaning that now the clipboard manager is the central point for all selections and is responsible for implementing correctly all the possible way for the clipboard to be requested.
Surprise, most clipboard manager who works like that is breaking applications in some kind of ways as they sucks to handle all the cases.

Greenclip started has a simple project and in order to not break other application clipboard (until 3.0-beta there is no image support for example) I designed it differently from other clipboard manager.
Instead of stealing the selection to other applications, greenclip is simply pooling for the content of the selection every 0.5 second and record only the changes.
By doing that applications still remain in charge of their selections and are free to implement all the cases they want to handle.
This means that greenclip will never break selections of other applications and never get outdated due to it failing to handle a new case that appeared in the wild.
On the other end, this design choice put limitation on what greenclip can do. It will never be able to handle large selection as pooling for them will burn your cpu, it will not be able to sync selection as greenclip does not own the selection until you request it to do so, ...

So to sum up, greenclip will not be able to do that as I designed the application in a conservative way that guarantee me that it will never break other clipboard's application, but this conservative choice put limitation on what it can do.
